### PR TITLE
My Site Dashboard: Tabs - Show Dynamic Cards on either tab

### PR DESCRIPTION
**Pull request by zwarm**
Parent #15989 

This PR adds the Dynamic Quick Start Cards to either tab depending on the initial screen (app setting) value.

Note: Only 1 approval is needed. Thanks

**To test:**
- Launch the app 
- Login in
- After site selection, select "Show me around" when prompted
- Navigate to Me -> App Settings 
- Note down which tab is the "initial screen" 
- Navigate to Me -> App Settings -> Debug Settings
- Enable `QuickStartDynamicCardsFeatureConfig` and restart the app
- Navigate to My Site
- ✅  Verify that Dynamic cards shows up "initial screen" tab
- Navigate back to Me -> App Settings 
- Set "initial screen" to the opposite tab
- Navigate back to My Site
- ✅  Verify that Dynamic cards shows up "initial screen" tab

## Regression Notes
1. Potential unintended areas of impact
Dynamic card doesn't show

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit testing

3. What automated tests I added (or what prevented me from doing so)
Add new tests to `MySiteViewModelTest`

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt
Comment (ashiagr) [inline WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt:2063]: Can we add two more tests to check that the dynamic card does not exist on the non-initial screen?

1. `given tabs enabled + site menu initial screen, when dashboard cards + items, then dynamic card not exists`
2. `given tabs enabled + dashboard initial screen, when site menu cards + items, then dynamic card not exists`

**Review by ashiagr — APPROVED** [review]
Works as described, good job, @zwarm! I just left one comment regarding tests. 
I'll approve the PR but not merge it to allow you to consider it.

It's so nice to see this project wrapping up. 🎉

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt
Comment (zwarm) [reply WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt:2063]: @ashiagr - Great idea. Thanks!! in [72edb6c](https://github.com/wordpress-mobile/WordPress-Android/pull/16277/commits/72edb6c381559c9a6c9dfbd992bedb6ba7c22530). I'll merge after CI finishes. :)

**Pull request merged**
